web/src/components/proxied-fetch.ts

```diff
@@ -65,14 +58,14 @@ export const cancellableFetch = <T>(
   const timeout = init?.timeout ?? 30 * 1000;
 
   if (timeout <= 0) {
-    return { request: () => fetchPromise, abort };
+    return fetchPromise;
   }
 
   const timeoutPromise = new Promise<T>((_resolve, reject) => {
     setTimeout(() => reject(new TimeoutError(url.toString(), timeout)), timeout);
   });
 
-  const request = () => Promise.race([fetchPromise, timeoutPromise]);
+  const request = Promise.race([fetchPromise, timeoutPromise]);
```

Comment:
We might need to keep this as it is as the thunk allows to start the timeout at the moment the request is executed rather than when the proxied fetch is called